code/game/objects/structures/aliens.dm

```diff
@@ -254,17 +254,24 @@
 	var/silent_removal = FALSE
 
 /obj/structure/alien/weeds/Initialize(mapload, node)
-	..()
+	. = ..()
 	linked_node = node
 	if(isspaceturf(loc))
-		qdel(src)
-		return
+		return INITIALIZE_HINT_QDEL
 	check_surroundings()
-	START_PROCESSING(SSobj, src)
+
 	RegisterSignal(linked_node, COMSIG_PARENT_QDELETING, PROC_REF(clear_linked_node))
+	RegisterSignal(loc, COMSIG_TURF_CHANGE, PROC_REF(on_turf_change))
+
+	if((istype(linked_node, /obj/structure/alien/resin/door)) || (istype(linked_node, /obj/structure/alien/resin/wall)))
+		return
+
+	if(get_dist(linked_node, src) > linked_node.node_range)
+		return
+
+	addtimer(CALLBACK(src, PROC_REF(initial_spread)), 10 SECONDS)
```

Comment:
This is going to perform even worse if its timer spam. See https://github.com/ParadiseSS13/Paradise/pull/18805

Reply:
it'll make 4 timers per change max. After that it'll remain dormant till a change happens again.
So mass destroying weeds without the root node will cause quite some timers to exist